maggy/ablation/ablationstudy.py

```diff
@@ -0,0 +1,233 @@
+class AblationStudy(object):
+    def __init__(self, training_dataset_name, training_dataset_version,
+                 label_name, **kwargs):
+        self.features = Features()
+        # TODO upgrade for Featurestore V2
+        # TODO call the featurestore and save the list of features of the dataset
+        self.model = Model()
+        self.hops_training_dataset_name = training_dataset_name
+        self.hops_training_dataset_version = training_dataset_version
+        self.label_name = label_name
+        self.custom_dataset_generator = kwargs.get('dataset_generator', False)
+
+    def to_dict(self):
+        """
+        Returns the ablation study configuration as a Python dictionary.
+        :return: A dictionary with ablation study configuration parameters as keys
+        (i.e. 'training_dataset_name', 'included_features', etc.)
+        :rtype: dict
+        """
+        ablation_dict = {
+            'training_dataset_name': self.hops_training_dataset_name,
+            'training_dataset_version': self.hops_training_dataset_version,
+            'label_name': self.label_name,
+            'included_features': list(self.features.included_features),
+            'included_layers': list(self.model.layers.included_layers),
+            'custom_dataset_generator': True if self.custom_dataset_generator else False,
+        }
+
+        return ablation_dict
+
+
+class Features(object):
+    # TODO type-checking for all the methods
+    def __init__(self):
+        self.included_features = set()  # TODO set or list?
+
+    def include(self, *args):
+        """
+        Add one or several features of the dataset to the ablation study. Features can be included one by one or
+        as a list of strings.
+        :param args: Strings or lists of strings, that should match feature names.
+        :type args: str or list
+        :return:
+        """
+        for arg in args:
+            if type(arg) is list:
+                for feature in arg:
+                    self._include_single_feature(feature)
+            else:
+                self._include_single_feature(arg)
+
+    def _include_single_feature(self, feature):
+        if type(feature) is str:
+            self.included_features.add(feature)  # TODO should check with the list retrieved from the featurestore
+        else:
+            raise ValueError("features.include() only accepts strings or lists of strings, "
+                             "but it received {0} which is of type '{1}'."
+                             .format(str(feature), type(feature).__name__))
```

Comment:
Maybe change indenting:
```suggestion
            raise ValueError(
                "features.include() only accepts strings or lists of strings, "
                "but it received {0} which is of type '{1}'."
                .format(str(feature), type(feature).__name__))
```